app/Services/Payroll/PayrollContextLocker.php

```
@@ -12,88 +12,295 @@
+    {
+        self::assertStage($lease, self::STAGE_PROFILES);
+        self::assertOwns($lease, $company);
+
+        if ((int) ($attributes['company_id'] ?? 0) !== $company->id) {
+            throw new LogicException('Owned payroll profile must belong to the locked company.');
+        }
+
+        $profile = WorkScheduleProfile::withoutCompanyScope()->create($attributes);
+        self::registerOwned($lease, collect([$profile]));
+
+        return $profile;
```

PHPStan (2.2.7)

[error] Method App\Services\Payroll\PayrollContextLocker::createOwnedProfile() should return App\Models\WorkScheduleProfile but returns Illuminate\Database\Eloquent\Model.


(return.type)